src/App.js

```diff
@@ -1,63 +1,152 @@
     }
+  }
+
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+
+  // Use useEffect to manage localStorage and persist todoList
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList)); // Save the todoList in localStorage when it changes
+    }
+  }, [todoList, isLoading]);
+
+
+  // Handle adding a new todo
+  async function handleAddTodo(newTodo) {
+    try {
+      const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
 
 
-    // Handle adding a new todo
-    function addTodo(newTodo) {
-      setTodoList([...todoList, newTodo]); // Update todoList state by adding the new todo
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          records: [{ fields: { title: newTodo.title } }],
+        }),
+      });
+
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+
+
+      const data = await response.json();
+
+
+      // Update the todoList state by adding the new todo
+      setTodoList([...todoList, { id: data.id, title: newTodo.title }]);
+    } catch (error) {
+      console.error('Error adding todo:', error.message);
     }
+  }
 
-    return (
-      <>
-        <AddTodoForm onAddTodo={addTodo} />
-        {isLoading ? <p>Loading...</p> : <TodoList todoList={todoList} onRemoveTodo={removeTodo} />}
-        
 
-      </>
-    );
+  // Use useEffect to simulate loading state and fetch data from localStorage with a 2 seconds delay
+  useEffect(() => {
+    setTimeout(() => {
+      const savedTodoList = JSON.parse(localStorage.getItem("savedTodoList")) || [];
+      setTodoList(savedTodoList);
+      setIsLoading(false);
+    }, 2000);
+  }, []); // Empty dependency array, so this effect runs once after initial render
+
+
+
+
+  // Define the removeTodo function to remove a todo item
+  function removeTodo(id) {
+    // Filter the todoList array to exclude the todo item with the specified id
+    const updatedTodoList = todoList.filter((todo) => todo.id !== id);
+
+
+    // Update the todoList state with the new array of todos
+    setTodoList(updatedTodoList);
   }
 
+
+ return (
+  <BrowserRouter>
+    <Routes>
+      {/* Define a Route for the root path ("/") */}
+      <Route
+        path="/"
+        element={
+          <AddTodoForm onAddTodo={handleAddTodo} /> // Render the AddTodoForm component with the handleAddTodo function
+        }
+      />
+      {/* Define a Route for the "/new" path */}
+      <Route
+        path="/new"
+        element={
+          isLoading ? <p>Loading...</p> : <TodoList todoList={todoList} onRemoveTodo={removeTodo} /> // Render either loading indicator or the TodoList component
```

Comment:
You're slightly ahead of the assignment with this new work =D

You'll definitely get to a point where you put more meaningful content into the `/new` `Route`, but in the meantime they want to make sure that it looks different to the default `/` route. This will make it easier in the meantime to test that the router is working!

```suggestion
          <h1>Add New Todo!</h1>
```